### ISSUE TITLE
[SECURITY ISSUE] asks does not validate TLS certificates by default

### ISSUE THREAD

**Issue by njsmith** [theelous3/asks#134]
asks versions 2.3.2 through 2.3.5 have TLS certificate validation disabled by default:

```
In [1]: import asks, trio, requests                                             

In [2]: requests.get("https://wrong.host.badssl.com")                           
[...]
SSLError: HTTPSConnectionPool(host='wrong.host.badssl.com', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError("hostname 'wrong.host.badssl.com' doesn't match either of '*.badssl.com', 'badssl.com'")))

In [3]: trio.run(asks.get, "https://wrong.host.badssl.com")                     
Out[3]: <Response 200 OK>
```

This makes all applications using `asks` vulnerable to man-in-the-middle attacks.

As a workaround, users can explicitly pass `ssl_context=ssl.create_default_context()`. For example:

```
In [7]: trio.run(partial(asks.get, "https://wrong.host.badssl.com", ssl_context=ssl.create_default_context()))             

SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: Hostname mismatch, certificate is not valid for 'wrong.host.badssl.com'. (_ssl.c:1056)
```

# Diagnosis

The problem comes from this line in `sessions.py`:

https://github.com/theelous3/asks/blob/b6f2becb973104392c6a66dbc1fcd8eb6d1403a3/asks/sessions.py#L76

If the user didn't pass in their own `SSLContext` object, then asks defaults to using one constructed as `ssl.SSLContext()`. This creates a context object with insecure defaults. Instead, it should be calling `ssl.create_default_context()` to get an object with secure defaults.

In addition, asks should add a test to confirm that connections with invalid certificates correctly raise an error.

**Comment by njsmith** [comment]
@theelous3 It'd be nice to get this fixed soon. What are you thinking? Is there anything I can do to help?

**Comment by theelous3** [comment]
Hey NJ! Sorry I've been swamped recently, haven't even had a chance to sit down and read issues.

This looks good! What a nice issue and PR. Thanks a mill.

**Comment by theelous3** [comment]
Released in latest ver.

**Comment by freis** [comment]
Hi all, using trio with the parameter on asks ssl_context=ssl.create_default_context() I am getting:
ssl.CertificateError: hostname doesn't match either of

How I should approach this?

**Comment by njsmith** [comment]
@freis This was fixed in asks 2.3.6, so if the reason you're passing `ssl_context=ssl.create_default_context()` is to work around this bug, then the best approach is to upgrade asks instead.

And if you're still having problems with asks 2.3.6 and its default settings, then maybe file a new bug?

(It's possible the answer is that the host you're connecting to simply doesn't have a valid certificate, so asks is correctly rejecting it. You can pass `verify=False` if you want to disable certificate checking, but of course that's less secure.)
